sklearn/covariance/_robust_covariance.py

```diff
@@ -643,7 +643,8 @@ def fit(self, X, y=None):
         random_state = check_random_state(self.random_state)
         n_samples, n_features = X.shape
         # check that the empirical covariance is full rank
-        if (linalg.svdvals(np.dot(X.T, X)) > 1e-8).sum() != n_features:
+        if (linalg.svdvals(np.dot(X.T, X),
```

Comment:
X is validated but because of the `np.dot` call I think the input may be non finite.
e.g. on my machine I have:
```
>>> import numpy as np
>>> X = np.array([1, 1e300])
>>> np.dot(X.T, X)
inf
```